Sources_v3/Workers/EventSender_Tests.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class EventSender_Tests: StressTestCase {
+    typealias ExtraData = DefaultDataTypes
+    
+    var webSocketClient: WebSocketClientMock!
+    var apiClient: APIClientMock!
+    var database: DatabaseContainerMock!
+    var time: VirtualTime!
+    var eventSender: EventSender<ExtraData>!
+    
+    override func setUp() {
+        super.setUp()
+        
+        webSocketClient = WebSocketClientMock()
+        apiClient = APIClientMock()
+        database = try! DatabaseContainerMock(kind: .inMemory)
+        
+        time = VirtualTime()
+        VirtualTimeTimer.time = time
+        
+        eventSender = EventSender(database: database, webSocketClient: webSocketClient, apiClient: apiClient)
+        eventSender.timer = VirtualTimeTimer.self
+    }
+    
+    override func tearDown() {
+        apiClient.cleanUp()
+        super.tearDown()
+    }
+    
+    func test_keystroke_startTypingAndStopAfterTimeout() throws {
```

Comment:
Don't be afraid to give it more descriptive names. For example:
`test_keystroke_sendsStartTyping_and_afterTimeoutStopTyping`